packages/at_secondary_server/lib/src/verb/handler/enroll_verb_handler.dart

```diff
@@ -72,68 +78,75 @@ class EnrollVerbHandler extends AbstractVerbHandler {
   }
 
   Future<void> _handleEnrollmentRequest(
-      HashMap<String, String?> verbParams,
+      EnrollParams enrollParams,
       currentAtSign,
       Map<dynamic, dynamic> responseJson,
       InboundConnection atConnection) async {
     if (!atConnection.getMetaData().isAuthenticated) {
-      var totp = verbParams['totp'];
-      if (totp == null || (await TotpVerbHandler.cache.get(totp)) == null) {
+      var totp = enrollParams.totp;
+      if (totp == null ||
+          (await TotpVerbHandler.cache.get(totp.toString()) == null)) {
         throw AtEnrollmentException(
             'invalid totp. Cannot process enroll request');
       }
     }
-    List<EnrollNamespace> enrollNamespaces = (verbParams['namespaces'] ?? '')
-        .split(';')
-        .map((namespace) =>
-            EnrollNamespace(namespace.split(',')[0], namespace.split(',')[1]))
-        .toList();
-    logger.finer('enrollNamespaces: $enrollNamespaces');
-
-    var enrollmentId = Uuid().v4();
-    var key = '$enrollmentId.$newEnrollmentKeyPattern.$enrollManageNamespace';
+    var enrollNamespaces = enrollParams.namespaces!;
+    var newEnrollmentId = _uuid.v4();
```

Comment:
Because you've made _uuid an instance variable, each enrollment would get the same enrollment id. Please revert to using Uuid().v4()

Reply:
And please add a unit test to verify that consecutive enrollments get unique ids

Reply:
done
https://github.com/atsign-foundation/at_server/pull/1496/commits/b22ea2028164aa54299758080d3b9033bf3a98e2